### PR TITLE
ORC-793: Deprecate the incorrect getInt method and add a new setInt within OrcConf

### PR DESCRIPTION
**Pull request by autumnust**
### What changes were proposed in this pull request?

There was two duplicated `getInt` method existed in `OrcConf` but no `setInt`. From the existing implementation it is obvious that the method `getInt` is a typo and it should `setInt` instead.

### How was this patch tested?

This patch is rather straightforward, running through existing unit tests.

### CODE REVIEW

File: java/core/src/java/org/apache/orc/OrcConf.java [hunk 1]
```diff
@@ -244,7 +244,7 @@ public int getInt(Configuration conf) {
     return getInt(null, conf);
   }
 
-  public void getInt(Configuration conf, int value) {
+  public void setInt(Configuration conf, int value) {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/OrcConf.java:247]: Wow. Nice catching.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you so much, @autumnust . 

Although this is a correct fix, Apache ORC follows semantic versioning rule. We cannot remove a public method like this at Apache ORC 1.7.x.

Instead, could you add a new function `public void setInt(Configuration conf, int value)` and add a deprecated annotation and description to the wrong old function, `public void getInt()`, please?

**Comment by autumnust** [comment]
> Thank you so much, @autumnust .
> 
> Although this is a correct fix and `void getInt` doesn't make senses, Apache ORC follows [semantic versioning](https://semver.org) rule. We cannot remove a public method like this at Apache ORC 1.7.x.
> 
> Instead, could you add a new function `public void setInt(Configuration conf, int value)` and add a deprecated annotation and description to the wrong old function, `public void getInt()`, please? And, the PR title and description should be revised accordingly.
> 
> cc @omalley , @pgaref , @wgtmac , @williamhyun

thanks for suggestions. @dongjoon-hyun  Do you think it is necessary to make the original incorrect `getInt` a noop?

File: java/core/src/java/org/apache/orc/OrcConf.java (hunk 1 above)
Comment (pgaref) [reply java/core/src/java/org/apache/orc/OrcConf.java:256]: Good catch indeed!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you so much, @autumnust .
Merged to master/1.6/1.5.

**Comment by dongjoon-hyun** [comment]
Also, thank you, @pgaref !

**Pull request closed without merging**